app/controllers/split_checkout_controller.rb

```diff
@@ -114,4 +130,15 @@ def advance_order_state
   def order_params
     @order_params ||= Checkout::Params.new(@order, params, spree_current_user).call
   end
+
+  # Update order state based on the step we are loading to avoid discrepancy between step and order
+  # state. We need to do this when moving back to a previous checkout step, the update action takes
+  # care of moving the order state forward.
+  def update_order_state
```

Comment:
I'm wondering why we're not comparing the `summary_step?` vs `order's state` in this method. 

This is just a question, I didn't spot any issues.

Reply:
You can't "go back" to the summary step, ie you can't go from details step to the summary one directly. You have to use the "Next" button, which will trigger an order update and move the order to the correct state.